roxie/roxiemem/roxiemem.cpp

```diff
@@ -7627,7 +7631,7 @@ class RoxieMemTests : public CppUnit::TestFixture
         memsize_t maxMemory = heapTotalPages * HEAP_ALIGNMENT_SIZE;
         memsize_t wasted = 0;
 
-        void * * pages = new void * [heapTotalPages];
+        std::unique_ptr<void *[]> pages(new void * [heapTotalPages]);
```

Comment:
I think this is the standard exception-safe way of protecting a new'd array.  Should we be using it elsewhere?